modules/otel-config-instrumentation.adoc

```diff
@@ -8,11 +8,25 @@
 
 The {OTELName} can inject and configure the OpenTelemetry auto-instrumentation libraries into your workloads. Currently, the project supports injection of the instrumentation libraries from Go, Java, Node.js, Python, .NET, and the Apache HTTP Server (`httpd`).
 
-Auto-instrumentation in OpenTelemetry refers to the capability where the framework automatically instruments an application without manual code changes. This enables developers and administrators to get observability into their applications with minimal effort and changes to the existing codebase.
+== How Autoinstrumentation works in the OpenTelemetry Operator
+
+Auto-instrumentation in {OTELOperator} refers to the capability where the framework automatically instruments an application without manual code changes. This enables developers and administrators to get observability into their applications with minimal effort and changes to the existing codebase.
+
+This process works as follows:
+
+1. **Container Modification**: The OpenTelemetry Operator modifies the container definition of your application. It adds the necessary instrumentation libraries for the specific language your application is written in (e.g., Java, Python, .NET, Node.js). This happens creating an init-container.
+   
+2. **Environment Variables**: The Operator sets the required environment variables in the application's runtime environment. These variables configure the autoinstrumentation libraries to collect traces, metrics, and logs and send them to the appropriate OpenTelemetry Collector or another telemetry backend.
+   
+3. **Automatic Instrumentation**: The injected libraries automatically instrument your application by hooking into known frameworks and libraries (e.g., web servers, database clients) to collect telemetry data. This happens without requiring any modifications to the application's source code.
+
+4. **Telemetry Data Flow**: Once the application is running with the injected instrumentation, it will automatically start generating telemetry data, which is sent to a designated OpenTelemetry Collector or an external OTLP endpoint for further processing.
```

Comment:
Question:
```
it [the application or instrumentation?] will automatically start generating telemetry data
```
?

Reply:
The automatically instrumented application

Reply:
Done.